### PR TITLE
rebase: update kube version to 1.23

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,34 +3,46 @@ module github.com/kubernetes-sigs/nfs-subdir-external-provisioner
...
 require (
-	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
-	k8s.io/api v0.18.0
-	k8s.io/apimachinery v0.18.0
-	k8s.io/client-go v0.18.0
+	github.com/golang/glog v1.0.0
+	k8s.io/api v0.23.4
+	k8s.io/apimachinery v0.23.4
+	k8s.io/client-go v0.23.4
 	k8s.io/kubernetes v1.18.0
```
Comment (pohly) [inline go.mod:10]: Shouldn't k8s.io/kubernetes also get updated?

File: go.mod [hunk 2]
```diff
@@ -3,34 +3,46 @@ module github.com/kubernetes-sigs/nfs-subdir-external-provisioner
...
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.23.4
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.23.4
+	k8s.io/kubectl => k8s.io/kubectl v0.23.4
+	k8s.io/kubelet => k8s.io/kubelet v0.23.4
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.23.4
+	k8s.io/metrics => k8s.io/metrics v0.23.4
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.23.4
 )
+
+replace k8s.io/component-helpers => k8s.io/component-helpers v0.23.4
```
Comment (pohly) [inline go.mod:38]: nit: it would be nicer to have these replace statements in the replace block above.

File: go.mod [hunk 3]
```diff
@@ -3,34 +3,46 @@ module github.com/kubernetes-sigs/nfs-subdir-external-provisioner
 go 1.14
 
 require (
-	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
-	k8s.io/api v0.18.0
-	k8s.io/apimachinery v0.18.0
-	k8s.io/client-go v0.18.0
-	k8s.io/kubernetes v1.18.0
+	github.com/golang/glog v1.0.0
```
Comment (pohly) [inline go.mod:6]: Just curious - why does the code still use glog?

Comment (humblec) [reply go.mod:6]: As you noted or mentioned earlier, this repo looks to be unmaintained and afaict, glog usage is side effect of the same. :( This PR bringing the dependency from kube v1.18 ( which is unsupported version) to v1.23 is an attempt to correct to the things.

File: go.mod (hunk 2 above)
Comment (humblec) [reply go.mod]: sure, done

**Review by pohly — APPROVED** [review]
/lgtm
/approve

File: go.mod (hunk 3 above)
Comment (pohly) [reply go.mod:6]: If it's not being maintained, then we should deprecate and archive it.

Comment (bswartz) [reply go.mod:6]: I agree with the sentiment, but I guarantee you people are using this CSI driver and would be upset if it went away. Maybe a call needs to go out putting this project up for adoption.

**Review by bswartz — APPROVED** [review]
/lgtm

File: go.mod (hunk 3 above)
Comment (kmova) [reply go.mod:6]: +1 to what @bswartz said. 

Personally am working on making time to maintain this repo. Would love to have some additional help!

**Pull request merged**